modules/hardware/laptop.nix

```diff
@@ -6,17 +6,18 @@
   lib,
   ...
 }: let
-  hwDefinition = import (./. + "/definitions/x1-${config.ghaf.hardware.generation}.nix");
+  cfg = config.ghaf.hardware.definition;
+  hwDefinition = import (./. + cfg.configFile);
   inherit (lib) mkOption types;
 in {
   imports = [
-    ../definition.nix
+    ./definition.nix
   ];
 
-  options.ghaf.hardware.generation = mkOption {
-    description = "Generation of the hardware configuration";
-    type = types.nullOr types.str;
-    default = null;
+  options.ghaf.hardware.definition.configFile = mkOption {
+    description = "Path to the hardware configuration file.";
+    type = types.str;
```

Comment:
null, no because it is required to be set to a valid file. As for being a path rather than a string. the handling of paths is somewhat different and will cause the file referenced as a path to be copied to the nix store. secondly the path is being created for th hwDefinition as it is relative to that place and not relative to where it is being called e.g. in the targets/laptop directory. so it behaves better as a string.

Reply:
it is the current way it is built. that is another refactoring to happen later, when we decide on config file vs maybe a module. yes the entire source is there, but path will add it again as a standalone path

Reply:
https://github.com/tiiuae/ghaf/pull/678/files#diff-f62a48ac04e1d3df38117cf98c13bcabfd4714ec99e6609701e1bd7bd6ec6a10R10